### ISSUE TITLE
Support go to definition

### ISSUE THREAD

**Issue by jvanbruegge** [dhall-lang/dhall-haskell#1577]
I am using the language server with vim and besides formatting, the feature of a language server I use the most is go to definition. Would be really nice to jump around in your code base

**Comment by sjakobi** [comment]
Go-to-import [should be supported](https://github.com/dhall-lang/dhall-haskell/tree/master/dhall-lsp-server#usage--features).

The idea is to extend this to bindings (`let`s, lambas)?!

**Comment by jvanbruegge** [comment]
I mean that when I have something like
```dhall
let utils = ../utils.dhall
in utils.doSomething "foo"
```
and I invoke `textDocument/definition` when having the cursor on `utils.doSomething` that it jumps into the utils file to the let binding that defines `doSomething`

**Comment by JohannesRudolph** [comment]
This would be a big improvement for sure, especially for those working with large dhall models (we do, dhall is the primary configuration model for our product).

**Comment by Gabriel439** [comment]
Yeah, this is in my list of things to get to.  Right now the higher priority thing I'm working on is fixing https://github.com/dhall-lang/dhall-haskell/issues/1511
